web_company_from_domain_name/models/ir_http.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Sunflower IT
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import models
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class IrHttp(models.AbstractModel):
+    _inherit = 'ir.http'
+
+    @classmethod
+    def _dispatch(cls):
+        """ Clear cache when a user switches to another domain name """
+        uid = request.session.uid
+        domain_name = request.httprequest.host.partition(":")[0]
+        cached_domain_name = request.registry['res.users']. \
+            get_cached_domain_name(uid, domain_name)
```

Comment:
you can just access `request.registry.cache`, and invent some unique key like `('ir.http', 'last_company', uid)` - if this exists and isn't your current company, you need to clear the cache

Reply:
Would that be better than this? I thought this was kind of neat; `get_cached_domain_name` has an `ormcache` decorator that defines just what you describe

Reply:
I find this a little bit indirect, but if you like your cached function better, keep it